src/sync/commits.ts

```diff
@@ -3,10 +3,60 @@ import { GitHubAPI } from "probot";
 import { Repository } from "../models/subscription";
 import GitHubClient from "../github/client/github-client";
 import { LoggerWithTarget } from "probot/lib/wrap-logger";
-import {getCommits as getCommitsQuery, getDefaultRef} from "../github/client/github-queries";
+import { getCommitsResponse, getCommitsQueryOctoKit, getDefaultRef }  from "../github/client/github-queries";
+import { booleanFlag, BooleanFlags } from "../config/feature-flags";
+
+// According to the logs, GraphQL queries sometimes fail because the "changedFiles" field is not available.
+// In this case we just try again, but without asking for the changedFiles field.
+const retryFetchCommits = async (logger: LoggerWithTarget, err, gitHubClient: GitHubClient, repoOwner: string, repoName: string, cursor?: string | number, perPage?: number): Promise<getCommitsResponse> => {
+	const changedFilesErrors = err.errors?.filter(e => e.message?.includes("The changedFiles count for this commit is unavailable"));
+
+	if (changedFilesErrors?.length) {
+		logger.info("retrying without changedFiles");
+		return await gitHubClient.getCommitsPage(false, repoOwner, repoName, perPage, cursor);
+	}
+	throw new Error(err);
```

Comment:
Not sure if this is correct, what I want to achieve is if the above error for changedFilesErrors doesn't exist then just to bubble up the error to be handled elsewhere

Reply:
Personally, I'd have this retry mechanism build into the `getCommitsPage` for everyone to use.